### PR TITLE
e2e/framework: fix root dir for in-process server

### PR DESCRIPTION
**Pull request by sttts**

### CODE REVIEW

**Comment by sttts** [comment]
Proof that it worked:

Bildschirmfoto 2022-07-04 um 09 32 13

File: test/e2e/framework/kcp.go [hunk 1]
```diff
@@ -236,7 +236,11 @@ func (c *kcpServer) Run(opts ...RunOption) error {
 
 	// run kcp start in-process for easier debugging
 	if runOpts.runInProcess {
-		serverOptions := options.NewOptions(".kcp")
+		rootDir := ".kcp"
+		if c.dataDir != "" {
+			rootDir = c.dataDir
+		}
+		serverOptions := options.NewOptions(rootDir)
```
Comment (s-urbaniak) [inline test/e2e/framework/kcp.go:243]: self-explanatory comment: this effectively moves the root directory to a temporary directory if running as in-process, maybe worth a small comment (not a blocker for merging).

**Comment by sttts** [comment]
Stop the bleeding.

**Pull request merged**